js/reference.js

```diff
@@ -1124,10 +1124,9 @@
                       "failed": null
                     });
                 }, function error(response) {
-                    var error = module.responseToError(response);
-                    return defer.reject(error);
+                    return defer.reject(module.responseToError(response, self));
```

Comment:
Since you've changed this, I feel like there's no point in having this reject callback. Just the catch-block should be sufficient.